recipe/0001-make-protobuf-linkable.patch

```diff
@@ -0,0 +1,159 @@
++                "third_party/google/protobuf/generated_message_reflection.cc",
++                "third_party/google/protobuf/generated_message_table_driven.cc",
++                "third_party/google/protobuf/generated_message_table_driven_lite.cc",
++                "third_party/google/protobuf/generated_message_util.cc",
++                "third_party/google/protobuf/implicit_weak_message.cc",
++                "third_party/google/protobuf/inlined_string_field.cc",
++                "third_party/google/protobuf/map.cc",
++                "third_party/google/protobuf/map_field.cc",
++                "third_party/google/protobuf/message.cc",
++                "third_party/google/protobuf/message_lite.cc",
++                "third_party/google/protobuf/parse_context.cc",
++                "third_party/google/protobuf/reflection_ops.cc",
++                "third_party/google/protobuf/repeated_field.cc",
++                "third_party/google/protobuf/repeated_ptr_field.cc",
++                "third_party/google/protobuf/text_format.cc",
++                "third_party/google/protobuf/unknown_field_set.cc",
++                "third_party/google/protobuf/wire_format.cc",
++                "third_party/google/protobuf/wire_format_lite.cc",
++                # Protobuf io
++                "third_party/google/protobuf/io/coded_stream.cc",
++                "third_party/google/protobuf/io/io_win32.cc",
++                "third_party/google/protobuf/io/strtod.cc",
++                "third_party/google/protobuf/io/tokenizer.cc",
++                "third_party/google/protobuf/io/zero_copy_stream.cc",
++                "third_party/google/protobuf/io/zero_copy_stream_impl.cc",
++                "third_party/google/protobuf/io/zero_copy_stream_impl_lite.cc",
++                # Protobuf stubs
++                "third_party/google/protobuf/stubs/common.cc",
++                "third_party/google/protobuf/stubs/int128.cc",
++                "third_party/google/protobuf/stubs/status.cc",
++                "third_party/google/protobuf/stubs/stringpiece.cc",
++                "third_party/google/protobuf/stubs/stringprintf.cc",
++                "third_party/google/protobuf/stubs/structurally_valid.cc",
++                "third_party/google/protobuf/stubs/strutil.cc",
++                "third_party/google/protobuf/stubs/substitute.cc",
++            ]
++        ),
+     ]
+ ]
+diff --git a/tools/pythonpkg/setup.py b/tools/pythonpkg/setup.py
+index 872ac2acd..f69cabd2c 100755
+--- a/tools/pythonpkg/setup.py
++++ b/tools/pythonpkg/setup.py
+@@ -91,6 +91,12 @@ if platform.system() == 'Windows':
+ if 'BUILD_HTTPFS' in os.environ:
+     libraries += ['crypto', 'ssl']
+     extensions += ['httpfs']
++if 'USE_SYSTEM_PROTOBUF' in os.environ:
++    if platform.system() == 'Windows':
++        libraries += ['libprotoc', 'libprotobuf', 'libprotobuf-lite']
++    else:
++        libraries += ['protoc', 'protobuf', 'protobuf-lite']
++    libraries += ['upb', 'address_sorting',]
```

Comment:
Weird, what's address_sorting for? And we need upb and protobuf?

Reply:
Mostly just copied what i did for grpcio which needed all of these.

Reply:
ah nm those come out of grpc, lemme fix